test/test_hash_state.py

```diff
@@ -20,29 +20,46 @@
 from girder.utility import hash_state
 import hashlib
 import pytest
-import sys
-import this
 
-testData = this.s.encode('utf8')
-chunkSize = len(testData) // 6
-chunks = [testData[i:i + chunkSize] for i in range(0, len(testData), chunkSize)]
-algorithms = hashlib.algorithms if sys.version_info <= (3, 2) else hashlib.algorithms_guaranteed
 
+@pytest.fixture
+def iterableBytes():
+    chunkSize = 256
+    chunkCount = 6
+    # A one-time iterable, with elements:
+    #  b'0000...'
+    #  b'1111...'
+    #  ...
+    iterable = (
+        str(chunkNum).encode('utf8') * chunkSize
+        for chunkNum in range(chunkCount)
+    )
 
-@pytest.mark.parametrize('alg', algorithms)
-def testSimpleHashing(alg):
-    canonical = hashlib.new(alg)
-    state = hash_state.serializeHex(hashlib.new(alg))
+    yield iterable
 
-    for chunk in chunks:
-        checksum = hash_state.restoreHex(state, alg)
-        assert canonical.hexdigest() == checksum.hexdigest()
-        assert canonical.digest() == checksum.digest()
 
-        canonical.update(chunk)
-        checksum.update(chunk)
-        state = hash_state.serializeHex(checksum)
+@pytest.mark.parametrize('algorithm', [
+    hashlib.md5,
+    hashlib.sha1,
+    hashlib.sha224,
+    hashlib.sha256,
+    hashlib.sha384,
+    hashlib.sha512
+])
```

Comment:
Mind adding the `ids` argument here for clearer test output?

Reply:
Refactored to accomplish the same thing.